lib/Sema/CSSimplify.cpp

```diff
@@ -8750,7 +8750,27 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyConformsToConstraint(
     return SolutionKind::Solved;
   }
 
-  return matchExistentialTypes(type, protocol, kind, flags, locator);
+  auto result = matchExistentialTypes(type, protocol, kind, flags, locator);
+
+  if (shouldAttemptFixes() && result.isFailure()) {
+    auto *loc = getConstraintLocator(locator);
+
+    if (loc->isLastElement<LocatorPathElt::InstanceType>())
+      loc = getConstraintLocator(loc->getAnchor(), loc->getPath().drop_back());
```

Comment:
Should we also handle nested cases? e.g

```swift
  func bar<T>(_ x: T) -> T.Type.Type { }
  func foo2(_ x: P.Type.Type.Type) { }
  foo2(bar(0))
```

Reply:
Eh, it just needs to unwrap in the loop, I'll fix that here.